Code/GraphMol/MolStandardize/Tautomer.h

```diff
@@ -40,27 +43,233 @@ inline int scoreTautomer(const ROMol &mol) {
+    typedef const ROMOL_SPTR &reference;
+    typedef std::input_iterator_tag iterator_category;
+
+    explicit const_iterator(const SmilesTautomerMap::const_iterator &it)
+        : d_it(it) {}
+    reference operator*() const { return d_it->second.tautomer; }
+    bool operator==(const const_iterator &other) const {
+      return (d_it == other.d_it);
+    }
+    bool operator!=(const const_iterator &other) const {
+      return !(*this == other);
+    }
+    const_iterator operator++(int) {
+      const_iterator copy(d_it);
+      ++*this;
+      return copy;
+    }
+    const_iterator &operator++() {
+      ++d_it;
+      return *this;
+    }
+
+   private:
+    SmilesTautomerMap::const_iterator d_it;
+  };
+  TautomerEnumeratorResult() : d_status(Completed) {}
+  TautomerEnumeratorResult(const TautomerEnumeratorResult &other)
+      : d_tautomers(other.d_tautomers),
+        d_status(other.d_status),
+        d_modifiedAtoms(other.d_modifiedAtoms),
+        d_modifiedBonds(other.d_modifiedBonds) {
+    fillTautomersItVec();
+  }
+  const const_iterator begin() const {
+    return const_iterator(d_tautomers.begin());
+  }
+  const const_iterator end() const { return const_iterator(d_tautomers.end()); }
+  size_t size() const { return d_tautomers.size(); }
+  bool empty() const { return d_tautomers.empty(); }
+  const ROMOL_SPTR &at(size_t pos) const {
+    PRECONDITION(pos < d_tautomers.size(), "index out of bounds");
+    return d_tautomersItVec.at(pos)->second.tautomer;
+  }
+  const ROMOL_SPTR &operator[](size_t pos) const { return at(pos); }
+  const boost::dynamic_bitset<> &modifiedAtoms() const {
+    return d_modifiedAtoms;
+  }
+  const boost::dynamic_bitset<> &modifiedBonds() const {
+    return d_modifiedBonds;
+  }
+  TautomerEnumeratorStatus status() const { return d_status; }
+  std::vector<ROMOL_SPTR> tautomers() const {
+    std::vector<ROMOL_SPTR> tautomerVec;
+    tautomerVec.reserve(d_tautomers.size());
+    std::transform(
+        d_tautomers.begin(), d_tautomers.end(), std::back_inserter(tautomerVec),
+        [](const SmilesTautomerPair &t) { return t.second.tautomer; });
+    return tautomerVec;
+  }
+  std::vector<ROMOL_SPTR> operator()() const { return tautomers(); }
+  std::vector<std::string> smiles() const {
+    std::vector<std::string> smilesVec;
+    smilesVec.reserve(d_tautomers.size());
+    std::transform(d_tautomers.begin(), d_tautomers.end(),
+                   std::back_inserter(smilesVec),
+                   [](const SmilesTautomerPair &t) { return t.first; });
+    return smilesVec;
+  }
+  const SmilesTautomerMap &smilesTautomerMap() const { return d_tautomers; }
+
+ private:
+  void fillTautomersItVec() {
+    for (auto it = d_tautomers.begin(); it != d_tautomers.end(); ++it) {
+      d_tautomersItVec.push_back(it);
+    }
+  }
+  // the enumerated tautomers
+  SmilesTautomerMap d_tautomers;
+  // status of the enumeration: did it complete? did it hit a limit?
```

Comment:
Minor point, this comment belongs to `d_status` not `d_tautomersItVec`

Reply:
Sure, I'll fix that.